plutus-contract/src/Plutus/Contract/Test/StateModel.hs

```diff
@@ -0,0 +1,176 @@
+-- This is a simple state modelling library for use with Haskell
```

Comment:
This is *also* standalone, right? Can we put it in the separate package and test it?

Reply:
Bigger question: why aren't we using `quickcheck-state-machine`?

Reply:
We looked at `quickcheck-state-machine` and chose not to use it for a few reasons:
- It's not actively maintained (cabal file says tested with `8.8.3`)
- It has a lot of complicated machinery for mixing symbolic and run-time data that we don't need
- We already had this module lying around from the quickcheck tutorial